### PR TITLE
gui comp: add SampleBackgroundOverlay for MIMAS

### PR DESCRIPTION
**Pull request by pieleric**
Shows the (theoretical) position of the samples based on the YAML
metadata into the view, as blue wireframe.

It could almost work for METEOR, but the metadata is in the SEM stage
(stage-bare) referential, while we need to have it in the FM referential
(stage).

### CODE REVIEW

File: src/odemis/gui/cont/tabs.py [hunk 1]
```diff
@@ -382,6 +381,11 @@ def __init__(self, name, button, panel, main_frame, main_data):
         panel.vp_secom_tl.canvas.remove_view_overlay(panel.vp_secom_tl.canvas.play_overlay)
         panel.vp_secom_tr.canvas.remove_view_overlay(panel.vp_secom_tr.canvas.play_overlay)
 
+        # Add a sample overlay to each view (if we have information)
+        if main_data.sample_centers:
```
Comment (nandishjpatel) [inline src/odemis/gui/cont/tabs.py:385]:
```suggestion
        if main_data.sample_centers and main_data.sample_radius:
```
maybe a better check

File: src/odemis/gui/comp/viewport.py [hunk 2]
```diff
@@ -650,6 +652,27 @@ def _set_fov_from_mpp(self):
...
+        the arguments have no effect.
+        sample: Dict of sample name -> center position (X/Y in m)
+        radius: the radius of a sample in m (all samples are shown with the same radius)
+        """
+        if not self._sample_overlay:
+            self._sample_overlay = overlay.world.SampleBackgroundOverlay(self.canvas, samples, radius)
+        self.canvas.add_world_overlay(self._sample_overlay)
+        wx.CallAfter(self.canvas.request_drawing_update)
+
+    def hide_sample_overlay(self):
```
Comment (nandishjpatel) [inline src/odemis/gui/comp/viewport.py:668]:
```suggestion
    def remove_sample_overlay(self):
```
a better naming?

File: src/odemis/gui/comp/overlay/world.py [hunk 3]
```diff
@@ -347,6 +351,67 @@ def set_icon(feature_icon):
...
+        self._labels = {}
+        for name in self._samples:
+            self._labels[name] = self.add_label(name,
+                                                align = wx.ALIGN_CENTRE_HORIZONTAL | wx.ALIGN_CENTER_VERTICAL,
+                                                colour=self._colour[:3],  # no transparency, to support .opacity
+                                                background=(0, 0, 0, 0.5)  # Black semi-transparent
+                                                )
+
+    def draw(self, ctx, shift=(0, 0), scale=1.0):
+        """ Draw the background image by displaying all circles. """
```
Comment (FCanberk) [inline src/odemis/gui/comp/overlay/world.py:379]: Would it be good to add parameter explanations in the doc string or do you think it is already self-explanatory?

**Review by nandishjpatel — APPROVED** [review]

File: src/odemis/gui/cont/tabs.py [hunk 4]
```diff
@@ -339,7 +339,6 @@ def get_display_priority(cls, main_data):
 AUTOFOCUS_BINNING = (8, 8)
 AUTOFOCUS_HFW = 300e-06  # m
 
-
 class LocalizationTab(Tab):
```
Comment (FCanberk) [inline src/odemis/gui/cont/tabs.py:342]: It's a bit of nitpicking but I think we should keep the line break before the class definition as PEP-8 states: https://peps.python.org/pep-0008/#blank-lines

**Pull request merged**